daemons/based/based_callbacks.c

```diff
@@ -1382,7 +1382,7 @@ cib_process_command(xmlNode *request, const cib__operation_t *operation,
     input = prepare_input(request, operation->type, &section);
 
     if (!pcmk_is_set(operation->flags, cib__op_attr_modifies)) {
-        rc = cib_perform_op(op, call_options, op_function, true, section,
+        rc = cib_perform_op(NULL, op, call_options, op_function, true, section,
```

Comment:
Pass `the_cib` here and below. Otherwise we're *always* skipping the feature set check.

Reply:
Never mind, I misread.

Reply:
Still probably good to do for future-proofing

Reply:
the_cib here is actually an `xmlNode *`, and one reason I went with NULL here is because `cib_process_command` would also need to take a `cib_t` as an argument, and it just kind of started to spiral out from there.  It seemed safer to go with NULL and continue performing the feature set check like we'd previously been doing.

Incidentally, maybe having cib, current_cib, and result_cib all as arguments to `cib_process_command` is too confusing.

Reply:
> the_cib here is actually an `xmlNode *`

Oh you're right, duh. The CIB manager won't have a `cib_t` object to connect to itself.

That makes sense to me.

> Incidentally, maybe having cib, current_cib, and result_cib all as arguments to `cib_process_command` is too confusing.

A lot of it is too confusing. It's actually a lot better than it was a year or so ago. I did a fair amount of cleanup and refactoring in based before and during my work on CIB transactions, so that I could figure out WTF was going on in there. It's still really complex, and it could be cleaned up and simplified a lot despite the inherent complexity that exists.

One thing that will help a ton is 3.0.0. There's a whole lot of legacy stuff in `based_callbacks.c` that exists only to handle rolling upgrades from Pacemaker 1.1.11 (the oldest version we support upgrades from). A lot of substantial modernization happened in 1.1.12. Which was too late.